### PR TITLE
Add bitsery 5.1.0

### PR DESCRIPTION
**Pull request by IceflowRE**
Specify library name and version:  **bitsery/5.1.0**
https://github.com/fraillt/bitsery

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Could you please tell me what is the best way to check for minimum cppstd? Across conan-center there are a ton of variations.

### CODE REVIEW

File: recipes/bitsery/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    license = "MIT"
+    settings = "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        tools.check_min_cppstd(self, "11")
```
Comment (uilianries) [inline recipes/bitsery/all/conanfile.py:21]:
```suggestion
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "11")
```

File: recipes/bitsery/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "bitsery-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include/bitsery", src=self._source_subfolder + "/include/bitsery")
```
Comment (uilianries) [inline recipes/bitsery/all/conanfile.py:30]:
```suggestion
        self.copy(pattern="*.h", dst=os.path.join("include", "bitsery"), src=os.path.join(self._source_subfolder, "include", "bitsery"))
```

File: recipes/bitsery/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+set(CMAKE_CXX_STANDARD 11)
```
Comment (uilianries) [inline recipes/bitsery/all/test_package/CMakeLists.txt:4]:
```suggestion
```

File: recipes/bitsery/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (uilianries) [inline recipes/bitsery/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 2.8.12)
```

File: recipes/bitsery/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+project(test_package)
+
+set(CMAKE_CXX_STANDARD 11)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/bitsery/all/test_package/CMakeLists.txt:10]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

File: recipes/bitsery/all/test_package/test_package.cpp [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  //it will use default configuration to setup all the nesessary steps
+  //and serialize data to container
+  auto writtenSize = bitsery::quickSerialization<OutputAdapter>(buffer, data);
+
+  //same as serialization, but returns deserialization state as a pair
+  //first = error code, second = is buffer was successfully read from begin to the end.
+  auto state = bitsery::quickDeserialization<InputAdapter>({buffer.begin(), writtenSize}, res);
+
+  assert(state.first == bitsery::ReaderError::NoError && state.second);
+  assert(data.fs == res.fs && data.i == res.i && data.e == res.e);
```
Comment (uilianries) [inline recipes/bitsery/all/test_package/test_package.cpp:48]:
```suggestion
```

File: recipes/bitsery/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class BitseryConan(ConanFile):
+    name = "bitsery"
+    description = "Header only C++ binary serialization library. It is designed around the networking requirements for real-time data delivery, especially for games."
+    topics = "serialization", "binary", "header-only"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/fraillt/bitsery"
+    license = "MIT"
+    settings = "compiler"
```
Comment (uilianries) [inline recipes/bitsery/all/conanfile.py:13]:
```suggestion
    settings = "compiler", "os"
```

File: recipes/bitsery/all/conanfile.py (hunk 1 above)
Comment (IceflowRE) [reply recipes/bitsery/all/conanfile.py:21]: Why is for C++11 no compiler check required?

**Pull request merged**